File: app/src/main/java/soup/movie/ui/settings/SettingsUiModel.java
package soup.movie.ui.settings;

import android.support.annotation.NonNull;

import java.util.List;

import soup.movie.data.soup.model.TheaterCode;
import soup.movie.ui.BaseUiModel;

interface SettingsUiModel extends BaseUiModel {

    class InProgress implements SettingsUiModel {

        InProgress() {
        }

        @Override
        public String toString() {
            return "SettingsUiModel.InProgress{}";
        }
    }

    class Data implements SettingsUiModel {

        private List<TheaterCode> theaterList;

        Data(@NonNull List<TheaterCode> theaterList) {
            this.theaterList = theaterList;
        }

        public List<TheaterCode> getTheaterList() {
            return theaterList;
        }

        @Override
        public String toString() {
            return "Data{" +
                    "theaterList=" + theaterList +
                    '}';
        }
    }
}

File: app/src/main/java/soup/movie/ui/settings/SettingsUiModel.kt
package soup.movie.ui.settings

import androidx.annotation.Keep
import soup.movie.data.model.Theater
import soup.movie.theme.ThemeOption

sealed class SettingsUiModel

@Keep
data class TheaterSettingUiModel(
    val theaterList: List<Theater>
) : SettingsUiModel()

@Keep
data class VersionSettingUiModel(
    val versionCode: Int,
    val versionName: String,
    val isLatest: Boolean
) : SettingsUiModel()

@Keep
data class ThemeSettingUiModel(
    val themeOption: ThemeOption
)
